modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -127,6 +136,34 @@ protected ReindexRequest buildRequest(RestRequest request) throws IOException {
         return internal;
     }
 
+    static RemoteInfo buildRemoteInfo(Map<String, Object> source) throws IOException {
+        @SuppressWarnings("unchecked")
+        Map<String, Object> remote = (Map<String, Object>) source.remove("remote");
+        if (remote == null) {
+            return null;
+        }
+        String username = extractString(remote, "username");
+        String password = extractString(remote, "password");
+        String hostInRequest = requireNonNull(extractString(remote, "host"), "[host] must be specified to reindex from a remote cluster");
+        Matcher hostMatcher = HOST_PATTERN.matcher(hostInRequest);
+        if (false == hostMatcher.matches()) {
+            throw new IllegalArgumentException(
+                    "[host] must be of the form [host], [host]:[port], [scheme]://[host], or [scheme]://[host]:[port]");
+        }
+        String scheme = hostMatcher.group("scheme");
+        if (scheme == null) {
+            scheme = password == null ? "http" : "https";
```

Comment:
They can still get `http` if they send the host as `http://somehost` or `http://something:9200`.